Range query  + VECTOR in JSON - update docs (#3189)
RediSearch/RediSearch: docs/docs/reference/Query_Syntax.md

* Selection of specific fields using the syntax `hello @field:world`.
* Numeric Range matches on numeric fields with the syntax `@field:[{min} {max}]`.
* Geo radius matches on geo fields with the syntax `@field:[{lon} {lat} {radius} {m|km|mi|ft}]`.

> * Range queries on vector fields (as of v2.6) with the syntax `@field:[VECTOR_RANGE {radius} $query_vec]`, where `query_vec` is given as a query param.

Reviewer: ```suggestion
* Range queries on vector fields (as of v2.6) with the syntax `@field:[VECTOR_RANGE {radius} $query_vec]`, where `query_vec` is given as a query parameter.
```

> * KNN queries on vector fields with or without pre-filtering (as of v2.4) with the syntax `{filter_query}=>[KNN {num} @field $query_vec]`

Reviewer: ```suggestion
* KNN queries on vector fields with or without pre-filtering (as of v2.4) with the syntax `{filter_query}=>[KNN {num} @field $query_vec]`.
```

## Vector Similarity search in query

It is possible to add vector similarity queries directly into the query language. There are two ways of performing vector similarity search:
1. Using a **range** query (as of v2.6.1) with the syntax of `@vector:[VECTOR_RANGE {radius} $query_vec]`, which filters the results to a given radius from a given query vector. The distance metric derives from the definition of @vector field in the index schema (e.g., Cosine, L2).

2. By running a **KNN** (K Nearest Neighbors) query on @vector field. The basic syntax is `"*=>[ KNN {num|$num} @vector $query_vec ]"`.

> It is also possible to run a Hybrid Query on filtered results. A Hybrid query allows the user to specify a filter criteria that ALL results in a KNN query must satisfy. The filter criteria can include any type of field (e.g., indexes created on both vectors and other values such as TEXT, PHONETIC, NUMERIC, GEO, etc)

Reviewer: ```suggestion
It is also possible to run a hybrid query on filtered results. A hybrid query allows the user to specify a filter criteria that all results in a KNN query must satisfy. The filter criteria can include any type of field (e.g., indexes created on both vectors and other values such as TEXT, PHONETIC, NUMERIC, GEO, etc.).
```

> The general syntax for hybrid query is `{some filter query}=>[ KNN {num|$num} @vector $query_vec]`, where `=>` separates filter query from vector KNN query. 

Reviewer: ```suggestion
The general syntax for hybrid query is `{some filter query}=>[ KNN {num|$num} @vector $query_vec]`, where `=>` separates the filter query from the vector KNN query. 
```

**Examples:**

> * `*=>[KNN 10 @vector_field $query_vec]` - Return 10 "nearest neighbors" entities in which `query_vec` is closest to the vector stored in `@vector_field`

Reviewer: ```suggestion
* `*=>[KNN 10 @vector_field $query_vec]` - Return 10 nearest neighbors entities in which `query_vec` is closest to the vector stored in `@vector_field`.
```

> * `@published_year:[2020 2022]=>[KNN 10 @vector_field $query_vec]` - Among entities published between 2020 and 2022, Return 10 "nearest neighbors" entities in which `query_vec` is closest to the vector stored in `@vector_field`

Reviewer: ```suggestion
* `@published_year:[2020 2022]=>[KNN 10 @vector_field $query_vec]` - Among entities published between 2020 and 2022, return 10 "nearest neighbors" entities in which `query_vec` is closest to the vector stored in `@vector_field`.
```

* `@vector_field:[VECTOR_RANGE 0.5 $query_vec]` - Return every entity for which the distance between the vector stored under its @vector_field and `query_vec` is at most 0.5, in terms of @vector_field distance metric.

> As of version 2.4, we allow KNN vector search to be used **once** in the query, while vector range filter can be used multiple times in a query (as of version 2.6). For more information on vector similarity syntax, see [Vector Fields](/redisearch/reference/vectors/#querying-vector-fields), "Querying vector fields" section.

Reviewer 2: consider adding a link also to the examples here

> As of version 2.4, we allow KNN vector search to be used **once** in the query, while, as of version 2.6, vector range filter can be used **multiple** times in a query. For more information on vector similarity syntax, see [Vector Fields](/docs/stack/search/reference/vectors/#querying-vector-fields), "Querying vector fields" and ["Examples"](/docs/stack/search/reference/reference/vectors/#vector-search-examples) section.

Reviewer: ```suggestion
As of version 2.4, the KNN vector search can be used **once** in the query, while, as of version 2.6, the vector range filter can be used **multiple** times in a query. For more information on vector similarity syntax, see [Querying vector fields](/docs/stack/search/reference/vectors/#querying-vector-fields), and [Vector search examples](/docs/stack/search/reference/reference/vectors/#vector-search-examples) sections.
```

> As of version 2.4, we allow KNN vector search to be used **once** in the query, while, as of version 2.6, vector range filter can be used **multiple** times in a query. For more information on vector similarity syntax, see [Vector Fields](https://redis.io/docs/stack/search/reference/reference/vectors/#querying-vector-fields), "Querying vector fields" and ["Examples"](https://redis.io/docs/stack/search/reference/reference/vectors/#vector-search-examples) section.

Reviewer 3: Revert also this full path to relative

The supported attributes are:

1. **$weight**: determines the weight of the sub-query or token in the overall ranking on the result (default: 1.0).
2. **$slop**: determines the maximum allowed "slop" (space between terms) in the query clause (default: 0).
3. **$inorder**: whether or not the terms in a query clause must appear in the same order as in the query, usually set alongside with `$slop` (default: false).

> 4. **$phonetic**: whether or not to perform phonetic matching (default: true). Note: setting this attribute on for fields which were not creates as `PHONETIC` will produce an error.

Reviewer: Please check the note sentence. It's unclear.

Author: I agree - but it's out of this PR scope... We can open an issue for that

As of v2.6.1, the query attributes syntax supports these additional attributes:

> * **$yield_distance_as**: specify the distance field name for later sorting by it and/or returning it, for clauses that yield some distance metric. Currently, supported for vector queries only (both KNN and range).   

Reviewer: ```suggestion
* **$yield_distance_as**: specify the distance field name for later sorting by it and/or returning it, for clauses that yield some distance metric. It is currently supported for vector queries only (both KNN and range).   
```

> * **vector query params**: pass optional params for [vector queries](/docs/stack/search/reference/vectors/#querying-vector-fields) in key-value format.

Reviewer: ```suggestion
* **vector query params**: pass optional parameters for [vector queries](/docs/stack/search/reference/vectors/#querying-vector-fields) in key-value format.
```
